File: app/src/main/java/com/akari/ppx/xp/hook/code/purity/FooterHook.java
package com.akari.ppx.xp.hook.code.purity;

import com.akari.ppx.xp.hook.code.SuperbHook;

import de.robv.android.xposed.XC_MethodReplacement;

import static com.akari.ppx.common.constant.Prefs.USE_NEW_FEED_FOOTER_STYLE;

public class FooterHook extends SuperbHook {
	@Override
	protected void onHook(ClassLoader cl) {
		hookMethod(USE_NEW_FEED_FOOTER_STYLE, "com.sup.superb.m_feedui_common.util.c", "a", XC_MethodReplacement.returnConstant(true));
	}
}

File: app/src/main/java/com/akari/ppx/xp/hook/assist/FooterHook.kt
@file:Suppress("unused")

package com.akari.ppx.xp.hook.assist

import com.akari.ppx.utils.hookAfterMethod
import com.akari.ppx.xp.Init.cl
import com.akari.ppx.xp.hook.SwitchHook

class FooterHook : SwitchHook("use_feed_footer_new_style") {
    override fun onHook() {
        "com.sup.superb.m_feedui_common.util.FeedCommonSettingsHelper\$useFeedFooterNewStyle\$2".hookAfterMethod(
            cl,
            "invoke"
        ) { param ->
            param.result = true
        }
    }
}